### PR TITLE
Vision foundation model

### PR DESCRIPTION
**Pull request by rjaincc**
# Description
Merging Image Foundation Model examples to main. 
Runs captured [here](https://loop.microsoft.com/p/eyJ3Ijp7InUiOiJodHRwczovL21pY3Jvc29mdC5zaGFyZXBvaW50LmNvbS8%2FbmF2PWN6MGxNa1ltWkQxaUlWOUlPV3RUV0RKelVrVkhURkZVUjAxeWJFTTRXVFpqVjBkNGMySkRhWEJDY25WaFUyVkxWRlJaTTAxWllVWnJWemxqUVhkU05qWkhkMVJZV2tzdGQyMG1aajB3TVRKU1VrZEdSVXhDUmtoU1JWSktSek15UmtkS1VrVkRVVUUwUzBNMk5GSkpKbU05Sm1ac2RXbGtQVEUlM0QiLCJyIjpmYWxzZX0sInAiOnsidSI6Imh0dHBzOi8vbWljcm9zb2Z0LnNoYXJlcG9pbnQuY29tLzpmbDovci9zaXRlcy80Yzg4OGNjMC1iMzZkLTRlMGQtOTRhZS00N2MzMGJkY2IwMTMvU2hhcmVkJTIwRG9jdW1lbnRzL0xvb3BBcHBEYXRhL1VudGl0bGVkJTIwcGFnZSUyMDEubG9vcD9kPXc4YTYwMjllYTgxNDg0YzM5OGM1ZWUxOGJmYjgzNzIyNiZjc2Y9MSZ3ZWI9MSZuYXY9Y3owbE1rWnphWFJsY3lVeVJqUmpPRGc0WTJNd0xXSXpObVF0TkdVd1pDMDVOR0ZsTFRRM1l6TXdZbVJqWWpBeE15WmtQV0loWDBnNWExTllNbk5TUlVkTVVWUkhUWEpzUXpoWk5tTlhSM2h6WWtOcGNFSnlkV0ZUWlV0VVZGa3pUVmxoUm10WE9XTkJkMUkyTmtkM1ZGaGFTeTEzYlNabVBUQXhNbEpTUjBaRlVFdEdSbEZKVlZORlFraEdSMGxaV0ZoQ1VsQTFXVWMwVWtjbVl6MGxNa1ltWm14MWFXUTlNU1poUFV4dmIzQkJjSEFtY0QwbE5EQm1iSFZwWkhnbE1rWnNiMjl3TFhCaFoyVXRZMjl1ZEdGcGJtVnlKbmc5SlRkQ0pUSXlkeVV5TWlVelFTVXlNbFF3VWxSVlNIaDBZVmRPZVdJelRuWmFibEYxWXpKb2FHTnRWbmRpTW14MVpFTTFhbUl5TVRoWmFVWm1VMFJzY2xVeFozbGpNVXBHVWpCNFVsWkZaRTVqYlhoRVQwWnJNbGt4WkVobFNFNXBVVEpzZDFGdVNqRlpWazVzVXpGU1ZWZFVUazVYVjBaSFlURmpOVmt3UmpOVmFsa3lVak5rVlZkR2NFeE1XR1IwWmtSQmVFMXNTbE5TTUZwR1ZFVktSMU5HU2taVmEzQklUWHBLUjFJd2NGTlNWVTVTVVZSU1RGRjZXVEJWYTJzbE0wUWxNaklsTWtNbE1qSnBKVEl5SlROQkpUSXlZVGhrTURrMk5ERXRPVEpoTnkwME5XVmpMV0l6WkRRdFlUUXdOR1psWmpFMllUVTBKVEl5SlRkRSIsInIiOmZhbHNlfSwiaSI6eyJpIjoiYThkMDk2NDEtOTJhNy00NWVjLWIzZDQtYTQwNGZlZjE2YTU0In19).

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md:18]: it should be azureml right?

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md:19]: same, please check.

**Review by gauravrajguru — CHANGES_REQUESTED** [review]
registry name

**Review by gauravrajguru — APPROVED** [review]

**Comment by skasturi** [comment]
Thanks for updating the code quality WF. 

Could you please also update the import model gate to include the updates?

**Review by nvijayrania — APPROVED** [review]

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md
Comment (nvijayrania) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:3]: Nit: we can have link to the actual component in azureml registry.

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md
Comment (nvijayrania) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md:57]: typo: AuzreML  --> AzureML

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md
Comment (nvijayrania) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_model_import_component.md:55]: typo: AuzreML  --> AzureML

**Review by skasturi — APPROVED** [review]

File: .github/workflows/code-quality.yml
Comment (skasturi) [inline .github/workflows/code-quality.yml:21]: While this will unblock this PR, let us revert it once this PR gets merged.

**Review by RSavage2 — APPROVED** [review]

File: .github/workflows/code-quality.yml
Comment (rjaincc) [reply .github/workflows/code-quality.yml:21]: Will raise the PR right after.

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:195]: nit: total amount of -> total number of

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:203]: worsens -> doesn't improve

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:223]: optimzer -> optimizer

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:227]: Output dir -> Output directory

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:232]: nit: links to the compute types?

**Review by vadthyavath — CHANGES_REQUESTED** [review]

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md
Comment (vadthyavath) [inline sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_model_import_component.md:33]: this link is not working

**Pull request merged**
